tensorflow/tools/dist_test/python/mnist_replica.py

```diff
@@ -52,29 +52,24 @@
 flags.DEFINE_boolean("download_only", False,
                      "Only perform downloading of data; Do not proceed to "
                      "session preparation, model definition or training")
-flags.DEFINE_integer("worker_index", 0,
-                     "Worker task index, should be >= 0. worker_index=0 is "
+flags.DEFINE_integer("task_index", 0,
+                     "Worker task index, should be >= 0. task_index=0 is "
                      "the master worker task the performs the variable "
                      "initialization ")
 flags.DEFINE_integer("num_workers", 2,
                      "Total number of workers (must be >= 1)")
-flags.DEFINE_integer("num_parameter_servers", 1,
-                     "Total number of parameter servers (must be >= 1)")
+flags.DEFINE_integer("num_gpu", 4,
+                     "Total number of gpus for each machine (must be >= 1")
```

Comment:
But if there are 2 workers both use `"/cpu:0"`, it may cause the problem, like this:
```
Traceback (most recent call last):
  File "mnist_replica.py", line 256, in <module>
    tf.app.run()
  File "/Users/Patient/anaconda/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 30, in run
    sys.exit(main(sys.argv))
  File "mnist_replica.py", line 195, in main
    global_step=global_step)
  File "/Users/Patient/anaconda/lib/python2.7/site-packages/tensorflow/python/training/supervisor.py", line 327, in __init__
    self._verify_setup()
  File "/Users/Patient/anaconda/lib/python2.7/site-packages/tensorflow/python/training/supervisor.py", line 863, in _verify_setup
    "their device set: %s" % op)
ValueError: When using replicas, all Variables must have their device set: name: "global_step"
```
So I have to verify if there is only 1 ps and 1 worker to use the `"/cpu:0"` when they set `num_gpus = 0`. Or do I need to set a new flag named `num_cpus` to handle this situation when they have more than 1 workers but don't have gpus.